data_diff/sqeleton/databases/mysql.py

```diff
@@ -127,17 +130,31 @@ def __init__(self, *, thread_count, **kw):
         self._args = kw
 
         super().__init__(thread_count=thread_count)
+        self.check_charset()
 
         # In MySQL schema and database are synonymous
         try:
             self.default_schema = kw["database"]
         except KeyError:
             raise ValueError("MySQL URL must specify a database")
 
+    def check_charset(self) -> None:
+        if logging.getLogger().level == logging.DEBUG:
+            try:
+                char_set_result = self.query("SELECT @@character_set_client;")
+                logger.debug(f"charset: {char_set_result.rows}")
+                collation_result = self.query("SELECT @@collation_connection;")
+                logger.debug(f"collation: {collation_result.rows}")
+            except Exception as ex:
+                logger.warning(f"Failed to check the charset: {ex}")
+
     def create_connection(self):
         mysql = import_mysql()
         try:
-            return mysql.connect(charset="utf8", use_unicode=True, **self._args)
+            conn = mysql.connect(charset="utf8mb4", use_unicode=True, **self._args)
+            conn.set_charset_collation(charset="utf8mb4", collation="utf8mb4_0900_ai_ci")
```

Comment:
This collation was added in MySQL 8.0.1 released sometime around 2018. We might expect much older versions for the typical migration/replication use cases for which data-diff is used (usually for the purpose of leaving those older versions). Should we make it so that the absence of this collation does not raise an error a few lines below?

And a second question: "ci" means "case insensitive". Does it make bisection by textual PKs, as well as `where`-filtering, case-insensitive by default, which might come as a surprise to users?

Reply:
@nolar I think I need to look into it more deeply, but from what I can tell, all of the default collations are _ci